Resolves #25, #28 and #31

sphericalice opened the pull request.

sphericalice committed:
Amending signature for ability_something()

sphericalice committed:
Fixes incorrect ability activation

Not 100% confirmed it's working perfectly but it's no longer having the issues I was having before

sphericalice committed:
Resolves #28

Touched commented on src/bs_command.c:
```diff
@@ -291,10 +291,9 @@ void set_b_x_callback(bxcb callback) {
-  team_index_by_side += CURRENT_BANK << 1;
-  current_team += (*team_index_by_side) * 100;
+  current_team = team;
+  if (CURRENT_BANK) current_team = enemy_team;
```
Does this not break doubles?

sphericalice committed:
Works with doubles

For some reason the player can't mega if the opponent does in doubles but I don't think that's related.
